adapters/zio-http/src/main/scala/caliban/ZHttpAdapter.scala

```diff
@@ -60,15 +64,21 @@ object ZHttpAdapter {
                    )
         in     = ZStream.fromQueueWithShutdown(queue)
         out    = pipe(in).map {
-                   case Right(output) => WebSocketFrame.Text(output.asJson.dropNullValues.noSpaces)
+                   case Right(output) => WebSocketFrame.Text(outputCodec.encode(output))
                    case Left(close)   => WebSocketFrame.Close(close.code, Some(close.reason))
                  }
         socket = Http
                    .collectZIO[WebSocketChannelEvent] {
                      case ChannelEvent(ch, UserEventTriggered(HandshakeComplete)) =>
                        out.runForeach(ch.writeAndFlush(_)).race(ch.awaitClose)
                      case ChannelEvent(_, ChannelRead(WebSocketFrame.Text(text))) =>
-                       ZIO.fromEither(decode[GraphQLWSInput](text)).flatMap(queue.offer)
+                       ZIO.fromEither {
+                         inputCodec.decode(text) match {
+                           case DecodeResult.Value(v)    => Right(v)
+                           case DecodeResult.Error(_, e) => Left(e)
+                           case _                        => Left(new Throwable("failed to decode input"))
```

Comment:
Do we need to customise the error message for all the different decoding failures here? There are the following failures that are currently not handled explicitly: `InvalidValue`, `Mismatch`, `Missing`, `Multiple`. The issue with these methods is that they don't wrap around a `Throwable`

Reply:
Or maybe `toString` the failure instead?

Reply:
Done